parent/pom.xml

```diff
@@ -1948,156 +1945,165 @@
                         <goals>
                            <goal>prepare-agent</goal>
                         </goals>
                      </execution>
                   </executions>
                   <configuration>
-                     <propertyName>jacocoAgent</propertyName>
+                     <propertyName>jacoco.agent.argLine</propertyName>
+                     <destFile>${dir.jacoco}/${project.artifactId}.exec</destFile>
+                     <excludes>
+                        <exclude>**/logging/**</exclude>
+                     </excludes>
                   </configuration>
                </plugin>
-
-
                <plugin>
                   <groupId>org.apache.maven.plugins</groupId>
-                  <artifactId>maven-surefire-plugin</artifactId>
-                  <configuration>
-                     <argLine combine.self="override">-Xmx2G ${jacocoAgent}</argLine>
-                     <parallel combine.self="override">tests</parallel>
-                     <threadCount combine.self="override">${infinispan.test.parallel.threads}</threadCount>
-                     <groups combine.self="override">${defaultTestGroup}</groups>
-                     <systemPropertyVariables combine.self="override">
-                           <infinispan.test.jgroups.protocol>${infinispan.test.jgroups.protocol}</infinispan.test.jgroups.protocol>
-                           <jgroups.bind_addr>127.0.0.1</jgroups.bind_addr>
-                           <java.net.preferIPv4Stack>true</java.net.preferIPv4Stack>
-                     </systemPropertyVariables>
-                     <trimStackTrace combine.self="override">false</trimStackTrace>
-                     <properties combine.self="override">
-                        <property>
-                           <name>listener</name>
-                           <value>${testNGListener}</value>
-                        </property>
-                     </properties>
-                  </configuration>
+                  <artifactId>maven-antrun-plugin</artifactId>
+                  <executions>
+                     <execution>
+                        <id>copy-generated-classes</id>
+                        <phase>test</phase>
+                        <goals>
+                           <goal>run</goal>
+                        </goals>
+                        <configuration>
+                           <target>
+                              <echo>Copying test classes to jacoco folder.</echo>
+                              <copydir src="${project.build.directory}/classes" dest="${jacoco.reportPath}/classes" />
+                           </target>
+                        </configuration>
+                     </execution>
+                  </executions>
                </plugin>
             </plugins>
          </build>
       </profile>
-
       <profile>
          <id>jacocoReport</id>
 
+         <activation>
+            <activeByDefault>false</activeByDefault>
```

Comment:
The flag `<activeByDefault>` is actually set to false by default [1]. So why do we need to set it up here explicitly? 

[1] https://maven.apache.org/ref/3.3.9/maven-settings/settings.html